src/python/pants/goal/context.py

```diff
@@ -37,18 +40,30 @@ class Context:
 
   # TODO: Figure out a more structured way to construct and use context than this big flat
   # repository of attributes?
-  def __init__(self, options, run_tracker, target_roots,
-               requested_goals=None, target_base=None, build_graph=None,
-               build_file_parser=None, build_configuration=None,
-               address_mapper=None, console_outstream=None, scm=None,
-               workspace=None, invalidation_report=None, scheduler=None):
+  def __init__(
+    self,
+    options: Options,
+    run_tracker: RunTracker,
+    target_roots,
```

Comment:
Surprisingly, this is not of type `base.target_roots.TargetRoots`. I didn't spend time figuring out the correct type hint nor did I add to the below because `self.context` is a beast and not worth spending too much time on with it being only V1.